mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -635,9 +657,9 @@ LogicalResult backwardWeightAtomicAdd(Conv2DBwdWeightOp op,
     auto gemmInputTransform =
         BottomUpTMBuilder::above(embedTransform, embedTransformAttr);
 
-    llvm::SmallVector<StringRef, 3> nonNHWDims = {"ci", "y", "x"};
+    llvm::SmallVector<StringRef, 3> nonNHWDims = {"ci", "0", "1"};
     matchUnderlyingOrder(nonNHWDims, gemmInputTransform);
-    llvm::SmallVector<StringRef, 3> nhwDims = {"n1", "ho", "wo"};
+    llvm::SmallVector<StringRef, 3> nhwDims = {"n1", "0o", "1o"};
```

Comment:
Future work will be generating this list with a `for (i : numberOfImageDimensions)` and some calls to `Twine`.

... Heck, while we're here, we might go ahead and extend `TransformMapBuilder` to let it work on `StringAttr`s transparently.